pennylane/ops/cv.py

```diff
@@ -796,33 +807,257 @@ class CatState(CVOperation):
+            -\hbar & 0 & 0\\
+            0 & 1 & 0\\
+            0 & 0 & 1
+        \end{bmatrix}
+
+    Args:
+        wires (Sequence[int] or int): the wire the operation acts on.
+    """
+    num_wires = 1
+    num_params = 0
+    par_domain = None
+
+    ev_order = 2
+
+    @staticmethod
+    def _heisenberg_rep(p):
+        hbar = 2
+        return np.diag([-0.5, 0.5 / hbar, 0.5 / hbar])
+
+
+class X(CVObservable):
+    r"""pennylane.expval.X(wires)
+    Returns the position expectation value in phase space.
+
+    This expectation command returns the value :math:`\braket{\x}`.
+
+    **Details:**
+
+    * Number of wires: 1
+    * Number of parameters: 0
+    * Observable order: 1st order in the quadrature operators.
+    * Heisenberg representation:
+
+      .. math:: d = [0, 1, 0]
+
+    Args:
+        wires (Sequence[int] or int): the wire the operation acts on.
+    """
+    num_wires = 1
+    num_params = 0
+    par_domain = None
+
+    ev_order = 1
+
+    @staticmethod
+    def _heisenberg_rep(p):
+        return np.array([0, 1, 0])
+
+
+class P(CVObservable):
+    r"""pennylane.expval.P(wires)
+    Returns the momentum expectation value in phase space.
+
+    This expectation command returns the value :math:`\braket{\p}`.
+
+    **Details:**
+
+    * Number of wires: 1
+    * Number of parameters: 0
+    * Observable order: 1st order in the quadrature operators.
+    * Heisenberg representation:
+
+      .. math:: d = [0, 0, 1]
+
+    Args:
+        wires (Sequence[int] or int): the wire the operation acts on.
+    """
+    num_wires = 1
+    num_params = 0
+    par_domain = None
+
+    ev_order = 1
+
+    @staticmethod
+    def _heisenberg_rep(p):
+        return np.array([0, 0, 1])
+
 
+class Homodyne(CVObservable):
+    r"""pennylane.expval.Homodyne(phi, wires)
+    Observable value of homodyne measurement in phase space.
 
-__all__ = [cls.__name__ for cls in all_ops]
+    This expectation command returns the value :math:`\braket{\x_\phi}`,
+    where :math:`\x_\phi = \x cos\phi+\p\sin\phi` is the generalised
+    quadrature operator.
+
+    **Details:**
+
+    * Number of wires: 1
+    * Number of parameters: 1
+    * Observable order: 1st order in the quadrature operators.
+    * Heisenberg representation:
+
+      .. math:: d = [0, \cos\phi, \sin\phi]
+
+    Args:
+        phi (float): axis in the phase space at which to calculate
+            the homodyne measurement.
+        wires (Sequence[int] or int): the wire the operation acts on.
+    """
+    num_wires = 1
+    num_params = 1
+    par_domain = "R"
+
+    grad_method = "A"
+    ev_order = 1
+
+    @staticmethod
+    def _heisenberg_rep(p):
+        phi = p[0]
+        return np.array([0, np.cos(phi), np.sin(phi)])  # TODO check
+
+
+class PolyXP(CVObservable):
+    r"""pennylane.expval.PolyXP(q, wires)
+    Observable value of a second-order polynomial observable.
+
+    Represents an arbitrary observable :math:`P(\x,\p)` that is a second order
```

Comment:
We'll have to be more careful to think about things like this that contain both first-order (expval) and second order (var) elements

Reply:
Do you mean in terms of a future `var()` function? I figured that if `PolyXP()` represented a first order observable, `var` could use analytic differentiation. If `PolyXP()` is 2nd order, it will have to fall back to finite-differences.

But it shouldn't effect `expval(PolyXP)` and `sample(PolyXP)` as far as I can tell.